posthog/src/main/java/com/posthog/PostHog.kt

```diff
@@ -475,6 +494,14 @@ public class PostHog private constructor(
         @Suppress("UNCHECKED_CAST")
         val groups = getPreferences().getValue(GROUPS) as? Map<String, Any>
 
+        val distinctId = this.distinctId
+        val anonymousId = this.anonymousId
+
+        if (distinctId.isBlank() || anonymousId.isBlank()) {
```

Comment:
@neilkakkar can you confirm that the decide API requires both fields non nullable non blank or just `distinctId` is required?

Reply:
is the right approach here to not make the call at all or still let it happen?
if we don't make the call, should we clean the feature flags state from memory/disk?

Reply:
```
generate_exception_response(
                            "decide",
                            "Decide requires a distinct_id.",
                            code="missing_distinct_id",
                            type="validation_error",
                            status_code=status.HTTP_400_BAD_REQUEST,
                        ),
```

https://github.com/PostHog/posthog/blob/88e5815abddb1d12e481a82738ced06af3f862e4/posthog/api/decide.py#L162-L176

looks like if you send `disable_flags` with the decide call then you can not send distinct_id otherwise you have to send something that to python is not None.

(so just whitespace would be accepted and you'd hope not actually match anyone 🙈)

Reply:
if you send `disable_flags` no flags are computed :P - the indent is probably hard to see, but disable_flags is for the entire flag calculation flow.

So, if you want flags in any case, we always need a distinctID. AnonID is optional

Reply:
We don't have the `advanced_disable_feature_flags` option on the Mobile SDKs yet so I'll not consider it, will just make the `anonymousId` optional here and only request if there's a valid `distinctId` at least.

Reply:
[`267f1a2` (#101)](https://github.com/PostHog/posthog-android/pull/101/commits/267f1a2e64be2473c8523fe2b4bfe9596b156d82)

Reply:
sounds good